### DISCUSSION TITLE
For a project with subfolders, is there a way to see subfolder name as field in Table view?

### DISCUSSION THREAD

**Discussion by ddanieltan** [marcusolsson/obsidian-projects#772] [Help]
I like the idea of having the top level folder as a project that can view all projects. And subfolders are individual projects for a specific category (E.g. personal, work, reading etc.).

But when I choose the top level project folder (with subfolders enabled), I see all the notes, but I don't have a useful field for each note's subfolder that I can use for grouping/sorting.

**Comment by Acylation** [comment]
The path field indicates which subfolder the note belongs to. Can playing around this suit your need?

**Comment by ramblinghamster** [comment]
I found one possible workaround for this, but there's a minor catch that didn't bug me too much. It could also be from my not knowing how to set Projects to behave differently however.

I have a demo structure I mocked up that looks like this. My `30 - Initiatives` folder is the hub for all subprojects. Every folder below I create as a subproject.
![image](https://github.com/marcusolsson/obsidian-projects/assets/10392263/4559c8a6-16c6-4e05-b488-8cad677360d0)

In the folder where I keep my templater templates, I made a `project-note` template with this at the front of the file:
```
<%* 
let subProject = ""
const currentFile = app.workspace.getActiveFile();
const pathArray= currentFile.path.split("/");

if (pathArray.length > 1) {
	// -1 will be the markdown file itself
	subProject=pathArray.at(-2)
}

await app.fileManager.processFrontMatter(currentFile, (frontmatter) => {
	frontmatter["subProject"] = subProject; 
});
%>
```

I whipped it together pretty hastily, but all I do is take the current path of the note, then find the folder that the note happened to be in. I then update the front matter to create a `subProject` property with that value.

In my templater options, I have this template configured to be the default note to be generated when I create notes under the folder where I house my projects:
![image](https://github.com/marcusolsson/obsidian-projects/assets/10392263/8234d1e2-497a-4081-8561-e569a04249e3)

The catch appears to be that if you are in the Projects table, and click the `Add Note` button, it will create a note without triggering the above templater option. 

If I instead go to my folder, right click on it, and then manually generate the note however, it does trigger my template and I ended up with a note with my `subProject` property.
![image](https://github.com/marcusolsson/obsidian-projects/assets/10392263/468ff807-56ae-43a4-8701-681a41ceb187)


In my Projects, I now just add a subProject column, and it shows the subfolder as the sub project.
![image](https://github.com/marcusolsson/obsidian-projects/assets/10392263/a132d533-c99a-4c17-aff2-68f0ac56a1cb)
